pilot/pkg/xds/xdsgen.go

```diff
@@ -60,11 +60,15 @@ func init() {
 	controlPlane = &corev3.ControlPlane{Identifier: string(byVersion)}
 }
 
-func (s *DiscoveryServer) findGenerator(typeURL string, con *Connection) model.XdsResourceGenerator {
+func (s *DiscoveryServer) findGenerator(typeURL string, con *Connection, delta bool) model.XdsResourceGenerator {
 	if g, f := s.Generators[con.proxy.Metadata.Generator+"/"+typeURL]; f {
 		return g
 	}
 
+	if g, f := s.Generators["delta/"+typeURL]; f && delta {
```

Comment:
nit: Check for delta/typeURL in generators only if it is delta

Reply:
well, it only returns the generator if its delta, are you saying we shouldn't even attempt access?